rmw_cyclonedds_cpp/src/rmw_node.cpp

```diff
@@ -3120,25 +3119,23 @@ get_endpoint_info_by_topic(
     handle_topic_endpoint_info_array_fini(endpoint_info, allocator);
     return RMW_RET_ERROR;
   }
+  size_t i = 0;
   for (const auto & epi : endpoint_info_vector) {
-    endpoint_info->info_array[endpoint_info->count] =
-      rmw_get_zero_initialized_topic_endpoint_info();
     if ((ret = set_rmw_topic_endpoint_info(
         node_cache,
         ppant_rd,
         allocator,
         epi,
         is_publisher,
-        endpoint_info->info_array[endpoint_info->count])) != RMW_RET_OK)
+        endpoint_info->info_array[i])) != RMW_RET_OK)
     {
-      endpoint_info->count++;
       handle_topic_endpoint_info_array_fini(endpoint_info, allocator);
       dds_delete(ppant_rd);
       return ret;
     }
-    endpoint_info->count++;
+    i++;
```

Comment:
@ivanpauno nit: use prefix increment

Reply:
The code base of this repository uses extensively post-increment when the return value is not used.
The google [style-guide](https://google.github.io/styleguide/cppguide.html#Preincrement_and_Predecrement) accepts both pre-increment and post-increment for scalar types.